### PR TITLE
Documentation for Quickstart, Pairing up, and Layouts and Plugins

### LINKED ISSUE #14: Adjust documentation for layouts
Layouts are a pretty big topic and needs to be added to the documentation.

### LINKED ISSUE #38: State of Documentation
Extracting from the docs, these things are missing so far:

#### General
- [x] #7 (solved in #57)
#### Start page (solved in #53)
- [x] Revise everything below bots.
- [ ] ~Add section on structure of database.~ 
- [x] Group technical stuff (events, commands, database, etc.) under one section.
#### Quickstart
- [ ] Double check. Is that true? Or is there another way to start with a fresh empty test room w/o killing the chat server?
- [ ] In the window of the chat bot, you will see some JSON structures that will be explained
- [ ] Provide example script that extracts the utterances in a parlAI style representation?
#### Pairing up participants and running multiple rooms (and bots)
- [ ] everything?
#### Layouts and Plugins
- [ ] #14
#### Tools (solved in #53)
- [x] Add usage instructions for tools added in #44 (Also mention, that the `lxml` package is needed)
#### Writing your own bots (solved in #53)
- [x] Disecting the minimal bot
- [x] Disecting the multi bot

What else is missing?

### PR DESCRIPTION
**Pull request by atuefekci**
I added the following parts:

**Quickstart**
- In the window of the chat bot, you will see some JSON structures that will be explained

**Pairing up participants and running multiple rooms (and bots)**
- everything

**Layouts and Plugins**
- some code-blocks weren't displayed correctly; fixed that

These are missing so far:

**Quickstart**
- Double check. Is that true? Or is there another way to start with a fresh empty test room w/o killing the chat server?
- Provide example script that extracts the utterances in a parlAI style representation?

closes #14 
closes #38

### CODE REVIEW

**Comment by atuefekci** [comment]
Regarding the two missing parts:

1. Double check. Is that true? Or is there another way to start with a fresh empty test room w/o killing the chat server?
- I couldn't figure out another way to start an empty testroom without killing the chat server..so can I remove this *TODO part* or maybe someone else has an idea?

2. Provide example script that extracts the utterances in a parlAI style representation?
- Unfortunately, I am clueless regarding this point with parlAI..

File: docs/slurk_multibots.rst [hunk 1]
```diff
@@ -3,4 +3,90 @@
...
+
+where \*token\* is to be replaced by the second token created in *Step 2*. The multi bot is also in the waiting room now. Check for confirmation in the terminal window:
+
+  ``MultiBot joined room: Waiting Room``
+ 
+All required bots are registered. In the next steps we need to log in the users.
+
+**Step 5:**
+
+Open http://127.0.0.1:5000/token again and fill in the following:
```
Comment (TimDiekmann) [inline docs/slurk_multibots.rst]: Same as above:
```suggestion
Open the server's token page, e.g. http://127.0.0.1:5000/token, again and fill in the following:
```

File: docs/slurk_multibots.rst [hunk 2]
```diff
@@ -3,4 +3,90 @@
...
+  - Room:    Waiting Room
+  - Task:    meetup
+  - Count:   2	
+  - Key:     *the secret key you've defined in the config.ini*
+
+Press "Generate tokens". You will see two tokens.
+
+**Step 6:**
+
+Open http://127.0.0.1:5000 in a new tab. Fill the first field with a username and the second field with one of the tokens created in *Step 5*. Press "Enter Chatroom". The user will join the waiting room. Next we need a second user.
```
Comment (TimDiekmann) [inline docs/slurk_multibots.rst]:
```suggestion
Open the server (http://127.0.0.1:5000 for this example) in a new tab. Fill the first field with a username and the second field with one of the tokens created in *Step 5*. Press "Enter Chatroom". The user will join the waiting room.

Next we need a second user.
```

File: docs/slurk_multibots.rst [hunk 3]
```diff
@@ -3,4 +3,90 @@
...
+.. _screenshot_void:
+.. figure:: slurk_waitingroom.png
+   :align: center
+   :scale: 60 %
+
+   A single user in the waiting room, waiting for another player to join
+
+**Step 7:**
+
+Open http://127.0.0.1:5000 again, but this time in a **different browser**. Fill the first field with a username and the second field with the other token from *Step 5*. Press "Enter Chatroom".
```
Comment (TimDiekmann) [inline docs/slurk_multibots.rst]:
```suggestion
Now open the page again, but this time in a **different browser**. Fill the first field with a username and the second field with the other token from *Step 5*. Press "Enter Chatroom".
```

File: docs/slurk_multibots.rst [hunk 4]
```diff
@@ -3,4 +3,90 @@
 Pairing up participants and running multiple rooms (and bots)
 =============================================================
 
-*Explain how to run the waiting room / pairup bot, and the multibot.*
+In the following steps you will learn how to let bots and human users enter the slurk waiting room where users will be paired up and put into a new task-specific chatroom along with bots. The bot handling this mechanism (moving human users and other bots from the waiting room to the taskroom) is the **pairup bot**. You can find it in the *sample_bots* directory.
```
Comment (TimDiekmann) [inline docs/slurk_multibots.rst]:
```suggestion
In the following steps you will learn how to let bots and human users enter the slurk waiting room where users will be paired up and put into a new task-specific chatroom along with bots. We provided an example bot handling this mechanism: The **pairup bot** moves human users and other bots from the waiting room to a taskroom. You can find it in the *sample_bots* directory.
```

Comment (TimDiekmann) [reply docs/slurk_multibots.rst]: We should explicitly state, that **pairup bot** is an example bot. This was also missinterpreted by many of the WG members: It just demonstrates, **how** moving users is working. This is btw. one of the reasons we want to deprecate the sample bots directory in the next version and remove it entirely in 2.0. cc #45

File: docs/slurk_multibots.rst [hunk 5]
```diff
@@ -3,4 +3,90 @@
...
+
+Open a terminal window and start slurk:
+
+  ``python chat.py``
+
+**Step 2:**
+
+Go to http://127.0.0.1:5000/token and fill in the following:
+
+  - Source:  *can be anything*
```
Comment (TimDiekmann) [inline docs/slurk_multibots.rst]: *can be anything* is only partly true as this fields exists for a reason. In this example, ofc, this holds. However in production, this is meant to identify e.g. different sessions.

File: docs/slurk_multibots.rst [hunk 6]
```diff
@@ -3,4 +3,90 @@
...
+   
+Press "Generate tokens".
+
+Now you have two tokens. One token will be used for the pairup bot (*Step 3*) and the other one for the multi bot (*Step 4*). The multi bot serves as an example bot here which will join the users in the taskroom later.
+   
+**Step 3:**
+
+Open a new terminal window and run the **pairup bot**:
+
+  ``python sample_bots/pairup_bot.py *token*``
```
Comment (TimDiekmann) [inline docs/slurk_multibots.rst]:
```suggestion

    .. code-block:: sh

        python sample_bots/pairup_bot.py *token*
```

File: docs/slurk_multibots.rst [hunk 7]
```diff
@@ -3,4 +3,90 @@
...
+ 
+  ``ConciergeBot joined room: Waiting Room``
+    
+*ConciergeBot* is the name of the pairup bot.
+
+**Step 4:**
+ 
+Open a third terminal window and run the **multi bot**:
+
+  ``python sample_bots/multi_bot.py *token*``
```
Comment (TimDiekmann) [inline docs/slurk_multibots.rst]:
```suggestion
    .. code-block:: sh

        python sample_bots/multi_bot.py *token*
```

File: docs/slurk_multibots.rst [hunk 8]
```diff
@@ -3,4 +3,90 @@
...
+   :align: center
+   :scale: 60 %
+
+   A single user in the waiting room, waiting for another player to join
+
+**Step 7:**
+
+Open http://127.0.0.1:5000 again, but this time in a **different browser**. Fill the first field with a username and the second field with the other token from *Step 5*. Press "Enter Chatroom".
+
+With two participants being present in the waiting room now the *pairup bot* will open a new chatroom. Both users will be removed from the waiting room and put into the newly created taskroom together with the *multi bot*. Now the users and the multi bot can interact with each other, e.g. you can use the commands **/new_image_private** and **/new_image_public** in the chatroom to see this in action.
```
Comment (TimDiekmann) [inline docs/slurk_multibots.rst]: It should be stated, that **/new_image_private** and **/new_image_public** is tight to our specific sample bot.

File: docs/slurk_quickstart.rst [hunk 9]
```diff
@@ -70,25 +70,32 @@ Now before we log onto the server in the way described above, let us create a bo
...
+- Key: *the secret key you've defined in the config.ini*
 
 That is, we create three tokens in one go.
 
 Copy the first one, and use it to do the following, in a new terminal window (`minimal.py` lives in `sample_bots`):
 
 - `python minimal.py` *<TOKEN>*
 
-In the terminal window where you started the chat server, you should see a message stating that `minimal bot has connected`. (In the window of the chat bot, you will see some JSON structures that will be explained TODO:ELSEWHERE.)
+In the terminal window where you started the chat server, you should see a message stating that `minimal bot has connected`. In the window of the chat bot, you will see some JSON structures. These structures contain several pieces of information, for example about the room and the user. They are needed to keep track of all events and to emit data to the right room or user. You can access all the information with the help of the predefined commands in the bots, e.g.:
```
[the remark this answers is not in the document]
Comment (atuefekci) [reply docs/slurk_quickstart.rst]: Thanks for the suggestions! I will apply them.

**Comment by TimDiekmann** [comment]
Review looks fine now. Could you mention me for merging, when _WIP_ is resolved?

**Comment by atuefekci** [comment]
> Review looks fine now. Could you mention me for merging, when _WIP_ is resolved?

Yes, I will do so. In fact, the only thing missing now is the part about parlAI.
We will discuss with @nilinykh  later if this is necessarily needed (at the moment) and then I will update you.

**Comment by TimDiekmann** [comment]
Thank you!

**Comment by atuefekci** [comment]
@TimDiekmann I talked to Nikolai and we came to the conclusion that the part about parlAI is not needed at the moment and that it could be removed. 
So in fact, the PR is ready to be merged now.

**Comment by TimDiekmann** [comment]
I wait for Travis to complete testing the doc generation and will merge it afterwards.

**Pull request merged**